pkgs/os-specific/linux/unscd/default.nix

```diff
@@ -0,0 +1,59 @@
+{ fetchurl, fetchpatch, stdenv, systemd, lib }:
+
+stdenv.mkDerivation rec {
+  pname = "unscd";
+  version = "0.54";
+
+  src = fetchurl {
+    url = "https://busybox.net/~vda/${pname}/nscd-${version}.c";
+    sha256 = "0iv4iwgs3sjnqnwd7dpcw6s7i4ar9q89vgsms32clx14fdqjrqch";
+  };
+
+  unpackPhase = ''
+    cp $src nscd.c
+    chmod u+w nscd.c
+  '';
+
+  patches = [
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/u/${pname}/${version}-1/debian/patches/change_invalidate_request_info_output";
+      sha256 = "17whakazpisiq9nnw3zybaf7v3lqkww7n6jkx0igxv4z2r3mby6l";
+    })
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/u/${pname}/${version}-1/debian/patches/support_large_numbers_in_config";
+      sha256 = "0jrqb4cwclwirpqfb6cvnmiff3sm2jhxnjwxa7h0wx78sg0y3bpp";
+    })
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/u/${pname}/${version}-1/debian/patches/no_debug_on_invalidate";
+      sha256 = "0znwzb522zgikb0mm7awzpvvmy0wf5z7l3jgjlkdpgj0scxgz86w";
+    })
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/u/${pname}/${version}-1/debian/patches/notify_systemd_about_successful_startup";
+      sha256 = "1ipwmbfwm65yisy74nig9960vxpjx683l3skgxfgssfx1jb9z2mc";
+    })
+    ./remove-old-socket.diff
```

Comment:
Where is this patch coming from? Can we fetch it from elsewhere, like the ones above?

Reply:
No, I wrote it.

Reply:
On a deeper look, it's not allowed to create `NSCD_SOCKET_OLD`, right?

Can you add a commit message to the patch (and export via `git format-patch`), so people see what this is about directly?

Also, the general refactoring might be upstreamable - or a version that handles not being able to create `NSCD_SOCKET_OLD` gracefully.

It's always preferrable to push patches upstream, so they eventually end up there, instead of only in nixpkgs. We really should only include them here if upstream rejected it, but it doesn't properly work in nixpkgs.

Reply:
It was planned to submit this change upstream. I will add the patch header as you requested.

Reply:
Thanks! Please also add a comment linking to the discussion with upstream, so people bumping the package can follow the discussion, to see if/how the patch got accepted.

Reply:
I don't think there is any proper upstream mailing list I could link to. I sent a patch to the author's personal email address.

Reply:
The author hasn't responded so far, I'm inclined to send it to the busybox mailing list and see if anyone can reach the author, or apply the patch.

Reply:
http://lists.busybox.net/pipermail/busybox/2021-June/088866.html

Reply:
I have changed the patch to a less invasive workaround and added an extended description, including a link to the upstream discussion. Feel free to re-open if you see any improvements that could be made :)